pom.xml

```diff
@@ -142,16 +142,22 @@
             <scope>provided</scope>
         </dependency>
         <!-- Testing -->
+        <dependency>
```

Comment:
Is this dependency needed? If I remove it locally your PR works just fine.

Reply:
Lines #145-150: The org.apache.sling.commons.osgi 2.4.2 dependency is required to resolve java.lang.NoClassDefFoundError: org/apache/sling/commons/osgi/Order test failures encountered when building with JDK 11 + Maven 3.9.6

Reply:
I don't see that when building locally with 

```
Apache Maven 3.9.9
Maven home: /usr/share/maven
Java version: 11.0.25, vendor: Oracle Corporation, runtime: /usr/lib64/jvm/java-11-openjdk-11
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "6.11.8-1-default", arch: "amd64", family: "unix"
```

but it might be caused by the fact that we now declare the dependency twice

```
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.apache.sling:org.apache.sling.scripting.javascript:jar:3.1.5-SNAPSHOT
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.apache.sling:org.apache.sling.commons.osgi:jar -> version 2.1.0 vs 2.4.2 @ line 145, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING]
```

I suggest you bump the existing Commons OSGi dependency to the version that is needed for the tests to pass and remove this one.